energymeters/energymeter.cpp

```diff
@@ -0,0 +1,206 @@
+            if (m_connected) {
+                m_connected = false;
+                emit connectedChanged(m_connected);
+            }
+            return;
+        } else {
+            if (!m_connected) {
+                m_connected = true;
+                emit connectedChanged(m_connected);
+            }
+        }
+        modbus_32_t value;
+        value.u = 0;
+        if (reply->result().length() == 1) {
+            value.u = static_cast<float>(reply->result().at(0));
+        } else if (reply->result().length() == 2) {
+            if (descriptor.dataType() == "float") {
+                value.u = static_cast<float>(static_cast<uint32_t>(reply->result().at(0)) << 16 | reply->result().at(1));
+            } else {
+                qCWarning(dcEnergyMeters()) << "Data type not supported" << descriptor.dataType();
+            }
+        } else {
+            return;
+        }
+
+        if (type == ModbusRegisterType::VoltageL1) {
+            if (descriptor.unit() == "mV")
+                value.f /= 1000.00;
+
+            emit voltageL1Received(value.f);
+        } else if (type == ModbusRegisterType::VoltageL2) {
+            if (descriptor.unit() == "mV")
+                value.f /= 1000.00;
+
+            emit voltageL2Received(value.f);
+        } else if (type == ModbusRegisterType::VoltageL3) {
+            if (descriptor.unit() == "mV")
+                value.f /= 1000.00;
+
+            emit voltageL3Received(value.f);
+        } else if (type == ModbusRegisterType::CurrentL1) {
+            if (descriptor.unit() == "mA")
+                value.f /= 1000.00;
+
+            emit currentL1Received(value.f);
+        } else if (type == ModbusRegisterType::CurrentL2) {
+            if (descriptor.unit() == "mA")
+                value.f /= 1000.00;
+
+            emit currentL2Received(value.f);
+        } else if (type == ModbusRegisterType::CurrentL3) {
+            if (descriptor.unit() == "mA")
+                value.f /= 1000.00;
+
+            emit currentL3Received(value.f);
+        } else if (type == ModbusRegisterType::ActivePower) {
+            if (descriptor.unit() == "kW") {
+                value.f *= 1000;
+            } else if (descriptor.unit() == "mW") {
+                value.f /= 1000.00;
+            }
+            emit activePowerReceived(value.f);
+        } else if (type == ModbusRegisterType::PowerFactor) {
+            emit powerFactorReceived(value.f);
+        } else if (type == ModbusRegisterType::Frequency) {
+            emit frequencyReceived(value.f);
+        } else if (type == ModbusRegisterType::EnergyConsumed) {
+            if (descriptor.unit() == "Wh") {
+                value.f /= 1000.00;
+            }
+            emit consumedEnergyReceived(value.f);
+        } else if (type == ModbusRegisterType::EnergyProduced) {
+            if (descriptor.unit() == "Wh") {
+                value.f /= 1000.00;
+            }
+            emit producedEnergyReceived(value.f);
+        }
```

Comment:
If you remain if this if else, what happens if if no condition where met? Should we warn about that?

Reply:
added warning